### PR TITLE
Lazy Images: Use calypso-build to build

### LINKED ISSUE #17562: Lazy Images: Improve JS handling
Originally reported in https://github.com/Automattic/jetpack/pull/17489#issuecomment-713520423 via @ockham regarding the Lazy Image package JS.

* We're only bundling code here (through Webpack), not transpiling it (through Babel). This means that statements like const end up in built files, as detected by the ES5 validation script.
* We're dropping our built files (*.min.js) into src/, rather than a separate location.
* We're adding built files to GH.
* We're including the Interaction Observer polyfill verbatim.

In order to fix these issue, [he] suggests

* using calypso-build (which takes care of transpilation, and by default drops files into a separate build/ directory, thus resolving items 1. and 2.)
* deleting built files from GH, and adding the build/ subdirectory to .gitignore in order to prevent that from happening in the future
* adding the Interaction Observer npm to package.json (rather than copying it verbatim)
* Revert https://github.com/Automattic/jetpack/pull/17561 to re-enable validation.

### PR DESCRIPTION
**Pull request by ockham**
Fixes #17562 -- or rather, most of it:

> * us[e] calypso-build (which takes care of transpilation, and by default drops files into a separate build/ [edit: should be `dist/`] directory, thus resolving items 1. and 2.)
> * delet[e] built files from GH, and adding the [`dist/`] subdirectory to .gitignore in order to prevent that from happening in the future
> * Revert https://github.com/Automattic/jetpack/pull/17561 to re-enable validation.

One item from #17562 isn't addressed by this PR:

> * add the Interaction Observer npm to package.json (rather than copying it verbatim)

I haven't yet found a really good way to do that, as we also need to build it, and include it as an external. We can address that in a subsequent iteration however.

#### Changes proposed in this Pull Request:
Use `calypso-build` to build the lazy-images package.

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

- Run `yarn build-production-packages`, and verify that it passes (it includes ES5 validation!).
- Verify that the Lazy images module still works (specifically, the built JS).

(I've tested a post with an image and the Lazy Images setting switched on and it seemed to work fine, but I'm not familiar enough with lazy images to be 100% certain that it's still working as before.)

#### Proposed changelog entry for your changes:
Use `calypso-build` to build the lazy-images package.

#### Note

In comments on #17562, it was brought up that we might need to include the built `js` files in GH in order for composer to be able to export this package. This would mean adding the `dist/` directory. Should we do that? Or can composer include a build step for JS?

### CODE REVIEW

**Comment by ockham** [comment]
Planning to pick this up tomorrow!

**Comment by ockham** [comment]
Should be ready for review.

File: package.json
Comment (brbrr) [inline package.json:22]: Any reason we use `npm` instead of `yarn`?

Comment (anomiex) [reply package.json:22]: Also the `cd ../../` shouldn't be necessary, yarn does its thing in a subshell.

If that's future-proofing for potentially having to build more packages in the future, it might be best to future-proof a little further and put the commands in a build script in `bin/`.

File: packages/lazy-images/package.json
Comment (anomiex) [inline packages/lazy-images/package.json:20]: What does this do that `"rm -rf dist"` doesn't?

Other than download and run code from the Internet, of course.

File: packages/lazy-images/package.json
Comment (anomiex) [inline packages/lazy-images/package.json:22]: Should the dependency on @automattic/calypso-build be declared below, so this works standalone in addition to when being called from ../../package.json?

File: packages/lazy-images/package.json
Comment (anomiex) [inline packages/lazy-images/package.json:25]: You added this as a dependency, which is good. But you're still using the local copy on line 22 and in src/lazy-images.php, which seems to defeat the purpose.

**Review by anomiex — DISMISSED** [review]

File: package.json
Comment (ockham) [reply package.json:22]: > Any reason we use `npm` instead of `yarn`?

Oversight on my part, apologies. I'll change to `npm` :+1: Thanks for spotting :+1:

File: packages/lazy-images/package.json
Comment (ockham) [reply packages/lazy-images/package.json:20]: It's cross-platform, so it should also work on Windows -- at least that's why we're using it in Calypso, which is where I copied it from :slightly_smiling_face:

File: packages/lazy-images/package.json
Comment (ockham) [reply packages/lazy-images/package.json:25]: Good spot! I was hoping to use the npm but then changed my mind, but forgot to remove this.

File: package.json
Comment (ockham) [reply package.json:22]: > Also the `cd ../../` shouldn't be necessary, yarn does its thing in a subshell.

Good point, I'll remove it :+1: 

> If that's future-proofing for potentially having to build more packages in the future, it might be best to future-proof a little further and put the commands in a build script in `bin/`.

I did it indeed for future-proofing! If we end up building more packages like this, I'd prefer using `lerna`, much like we do in Calypso, since it allows running the same command across a number of packages. (That would also allow us dropping the `cd packages/lazy-images/` here.) See e.g. https://github.com/Automattic/wp-calypso/blob/8b3ef578098c061eb6838c516761822908e911cb/package.json#L95, which basically runs the `prepare` script across all packages listed in [`lerna.json`](https://github.com/Automattic/wp-calypso/blob/8b3ef578098c061eb6838c516761822908e911cb/lerna.json).

File: packages/lazy-images/package.json
Comment (ockham) [reply packages/lazy-images/package.json:22]: Yeah, makes sense! I'll add it as a devDependency.

File: package.json
Comment (ockham) [reply package.json]: 29b950f5d9d0fee9b217008cfbb0b3ea20b7b657

Comment (ockham) [reply package.json]: Aaand f9c48b83b to use `yarn` instead of `npm run`.

File: packages/lazy-images/package.json
Comment (anomiex) [reply packages/lazy-images/package.json:20]: I'm not sure how much of Jetpack's stuff is intended to be run on Windows, versus gets run inside a Linux docker container anyway. I do see that the "clean" scripts in Jetpack's package.json all just use `rm -rf`.

If you are going to use a node thing, IMO it would be better to just declare it as a dependency instead of having npx download an arbitrary version at runtime.

**Comment by ockham** [comment]
Just noticed that `intersection-observer` doesn't need transpilation. Will copy it to `dist/` instead.

File: packages/lazy-images/package.json
Comment (ockham) [reply packages/lazy-images/package.json:25]: Ended up with a solution that actually uses the file from the npm package :tada:

File: packages/lazy-images/package.json
Comment (ockham) [reply packages/lazy-images/package.json:20]: Happy to use `rm -rf`, if that's what JP does for everything else :+1:

**Comment by ockham** [comment]
Thanks @brbrr and @anomiex for reviewing! I think I've addressed all feedback. One remaining question:

> In comments on #17562, it was brought up that we might need to include the built `js` files in GH in order for composer to be able to export this package. This would mean adding the `dist/` directory. Should we do that? Or can composer include a build step for JS?

**Review by anomiex — COMMENTED** [review]
> In comments on #17562, it was brought up that we might need to include the built `js` files in GH in order for composer to be able to export this package. This would mean adding the `dist/` directory. Should we do that? Or can composer include a build step for JS?

Composer itself probably can't include a build step for JS. It seems to use whatever you get from e.g. https://github.com/Automattic/jetpack-lazy-images/archive/v1.2.0.zip. Can you have a build step for those Github zip downloads? Or can whatever mirrors the code from this monorepo into https://github.com/Automattic/jetpack-lazy-images/ run your build step? I don't know.

Or can we do release branches that have the files checked in even if they're not checked into master, and live with the composer package not working if someone tries to use "dev-master" or anything else that's not one of those release branches?

Note that you're also depending on node_modules/intersection-observer/intersection-observer.js, so that would also have to be checked into the repo if you can't figure out how to do a build step.

**Comment by ockham** [comment]
> > In comments on #17562, it was brought up that we might need to include the built `js` files in GH in order for composer to be able to export this package. This would mean adding the `dist/` directory. Should we do that? Or can composer include a build step for JS?
> 
> Composer itself probably can't include a build step for JS. It seems to use whatever you get from e.g. https://github.com/Automattic/jetpack-lazy-images/archive/v1.2.0.zip.

Ah, I wasn't even aware of that repo, thanks for bringing that up!

> Can you have a build step for those Github zip downloads? 

We can have something like that, yeah: A GitHub action that's triggered by adding a new tag and then runs the build step and creates a release with the resulting zip attached as an asset. (Here's an example of what that looks like: https://github.com/WordPress/gutenberg/pull/19626 and https://github.com/ockham/gutenberg/releases/tag/untagged-91af46b87ff343e447c9) However, the resulting file will have a somewhat different URL. (The one your reference is GitHub's automatically provided zip of the archive contents, whereas we'll have to attach our own asset to the release.) Is it possible to change that URL wherever it's used to fetch that zip?

> Or can whatever mirrors the code from this monorepo into https://github.com/Automattic/jetpack-lazy-images/ run your build step? I don't know.

I think the GH action based approach is cleaner -- it runs the build step when it should be conceptually run, at the time of building a release.

> Or can we do release branches that have the files checked in even if they're not checked into master, and live with the composer package not working if someone tries to use "dev-master" or anything else that's not one of those release branches?

:thinking: Possibly, but overall, that seems a bit messier...

> Note that you're also depending on node_modules/intersection-observer/intersection-observer.js, so that would also have to be checked into the repo if you can't figure out how to do a build step.

Good point, thanks for bringing that up. Ideally, I'd like to avoid checking a `node_modules/` dir into GH. We could probably run `yarn` in the lazy-images directory (to install that dependency) when running the install in the top-level dir or so. (Akin to what lerna does; Calypso should have some precedent for this sort of thing, I'll look into it.)

**Comment by jeherve** [comment]
> live with the composer package not working if someone tries to use "dev-master" or anything else that's not one of those release branches?

I don't think that's an option, since we use `dev-master` ourselves in the Jetpack repo, when building the Jetpack plugin from `master` or from a branch for testing.

**Comment by anomiex** [comment]
> > live with the composer package not working if someone tries to use "dev-master" or anything else that's not one of those release branches?
> 
> I don't think that's an option, since we use `dev-master` ourselves in the Jetpack repo, when building the Jetpack plugin from `master` or from a branch for testing.

Within Jetpack itself we don't download `dev-master` from packagist, though, do we? It's supposed to symlink the version in the monorepo, and the required `yarn build` there also builds the package.

**Comment by jeherve** [comment]
Oh right, good point! Ultimately we do always end up building it when in Jetpack.

**Review by jeherve — CHANGES_REQUESTED** [review]
The build appears to be failing right now.

```
ERROR in ./src/js/lazy-images.js
Module build failed (from ./node_modules/thread-loader/dist/cjs.js):
Thread Loader (Worker 0)
[BABEL] jetpack/packages/lazy-images/src/js/lazy-images.js: Could not find plugin "proposal-class-properties". Ensure there is an entry in ./available-plugins.js for it. (While processing: "jetpack/packages/lazy-images/node_modules/@babel/preset-env/lib/index.js")
````

**Comment by ockham** [comment]
> > Can you have a build step for those Github zip downloads?
> 
> We can have something like that, yeah: A GitHub action that's triggered by adding a new tag and then runs the build step and creates a release with the resulting zip attached as an asset. (Here's an example of what that looks like: [WordPress/gutenberg#19626](https://github.com/WordPress/gutenberg/pull/19626) and https://github.com/ockham/gutenberg/releases/tag/untagged-91af46b87ff343e447c9) However, the resulting file will have a somewhat different URL. (The one your reference is GitHub's automatically provided zip of the archive contents, whereas we'll have to attach our own asset to the release.) Is it possible to change that URL wherever it's used to fetch that zip?

I can (tentatively) answer that now, after a bit of research: It seems that it's Composer itself (well, packagist) that fetches from that URL, and unfortunately, it's apparently not possible to modify that to pick up a custom asset from a release instead of the 'vanilla' zipball. See https://github.com/composer/packagist/issues/903 where this was requested for packagist.

> > Or can whatever mirrors the code from this monorepo into https://github.com/Automattic/jetpack-lazy-images/ run your build step? I don't know.
> 
> I think the GH action based approach is cleaner -- it runs the build step when it should be conceptually run, at the time of building a release.

We could however run the GH action inside of the Jetpack repo (rather than the mirror repo) whenever a new `lazy-images` release is tagged. We would however need to modify the sync job accordingly, in order to use the release asset, rather than a code snapshot. (I'm not too familiar with all this, but I think that p1HpG7-6Xk-p2 provides a good starting point.)

This sounds however like a bit too much effort for a PR like this that's meant to bring about a relatively simple change, so it's probably worth reconsidering other options at this point.

> > Or can we do release branches that have the files checked in even if they're not checked into master, and live with the composer package not working if someone tries to use "dev-master" or anything else that's not one of those release branches?
> 
>  Possibly, but overall, that seems a bit messier...

So maybe this could be an option to reconsider, but I'm not familiar enough with JP's release process. Could you elaborate how those branches are used and what would need to go where at what point? :sweat_smile: 

---

Overall, I haven't been able to commit a lot of time to this PR lately, since I'm mostly busy with Gutenberg and WP.com integration issues these days. @anomiex and @jeherve -- Would it be possible for y'all that you take this PR over from here? I'm still happy to provide guidance with regard to `calypso-build` or GitHub actions, but it seems like the missing pieces for this PR are more in the Jetpack release and Composer realm that I'm probably less familiar with than y'all are :blush:

**Comment by anomiex** [comment]
Ok, I think I see the path forward here.

A commit in the monorepo runs the workflow defined in .github/workflows/update-package.yml to push the commits to the mirror repo. The main work there is done by .github/actions/update-package/update-package.sh. And Github Actions even includes yarn already, so we can very easily run a build step in the middle of that. I've just pushed 1d7ffab6de4ca70e77f9099496455eabe71efb45 which makes that happen; details are open to discussion of course.

@ockham There is one problem, though: When I try `yarn build` in the package, it doesn't work.

<details>
<summary>I'm doing this in a docker environment, set up like this:</summary>

Run docker with `docker run --rm -it ubuntu:18.04`, then run the following inside it:
```
# Install tools in the generic Ubuntu image.
apt-get update && apt-get -y install git npm
npm install -g yarn

# Clone the patch.
git clone --branch=update/lazy-images-build-system --depth=1 https://github.com/Automattic/jetpack.git
cd jetpack

# Here's what the update-package action does, simplified.
BASE=$(pwd)
NAME=lazy-images
CLONE_DIR="__${NAME}__clone__"

git clone --depth 1 https://github.com/automattic/jetpack-$NAME.git $CLONE_DIR
cd $CLONE_DIR

find . | grep -v ".git" | grep -v "^\.*$" | xargs rm -rf
cp -r $BASE/packages/$NAME/. .

$BASE/.github/actions/update-package/build-$NAME.sh
```
</details>

<details>
<summary>The build fails with the following output:</summary>

```yarn install v1.22.10
info No lockfile found.
[1/4] Resolving packages...
warning @automattic/calypso-build > @babel/cli > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning @automattic/calypso-build > node-sass > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @automattic/calypso-build > node-sass > node-gyp > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @automattic/calypso-build > @babel/cli > chokidar > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
warning @automattic/calypso-build > babel-jest > @jest/transform > jest-haste-map > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
warning @automattic/calypso-build > jest-config > jest-environment-jsdom > jsdom > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @automattic/calypso-build > jest-emotion > css > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning @automattic/calypso-build > jest-emotion > css > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning @automattic/calypso-build > node-sass > request > har-validator@5.1.5: this library is no longer supported
warning @automattic/calypso-build > webpack > watchpack > watchpack-chokidar2 > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning @automattic/calypso-build > jest-config > jest-environment-jsdom > jsdom > left-pad@1.3.0: use String.prototype.padStart()
warning @automattic/calypso-build > jest-config > jest-environment-jsdom > jsdom > request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
warning @automattic/calypso-build > jest-emotion > css > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
[2/4] Fetching packages...
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@2.1.3: The platform "linux" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@2.1.3: The engine "node" is incompatible with this module. Expected version "^8.16.0 || ^10.6.0 || >=11.0.0". Got "8.10.0"
[3/4] Linking dependencies...
warning " > @automattic/calypso-build@6.1.0" has unmet peer dependency "enzyme@^3.11.0".
warning "@automattic/calypso-build > enzyme-adapter-react-16@1.15.1" has unmet peer dependency "enzyme@^3.0.0".
warning "@automattic/calypso-build > enzyme-adapter-react-16@1.15.1" has unmet peer dependency "react@^16.0.0-0".
warning "@automattic/calypso-build > enzyme-adapter-react-16@1.15.1" has unmet peer dependency "react-dom@^16.0.0-0".
warning "@automattic/calypso-build > enzyme-to-json@3.4.3" has unmet peer dependency "enzyme@^3.4.0".
warning "@automattic/calypso-build > jest-enzyme@7.1.2" has unmet peer dependency "enzyme@>=3.4.0".
warning "@automattic/calypso-build > jest-enzyme@7.1.2" has unmet peer dependency "jest@>=22.0.0".
warning "@automattic/calypso-build > jest-enzyme > enzyme-to-json@3.6.1" has unmet peer dependency "enzyme@^3.4.0".
warning "@automattic/calypso-build > enzyme-adapter-react-16 > enzyme-adapter-utils@1.13.1" has unmet peer dependency "react@0.13.x || 0.14.x || ^15.0.0-0 || ^16.0.0-0".
warning "@automattic/calypso-build > enzyme-adapter-react-16 > react-test-renderer@16.14.0" has unmet peer dependency "react@^16.14.0".
warning "@automattic/calypso-build > jest-enzyme > enzyme-matchers@7.1.2" has unmet peer dependency "enzyme@>=3.4.0".
warning "@automattic/calypso-build > jest-enzyme > jest-environment-enzyme@7.1.2" has unmet peer dependency "enzyme@3.x".
warning "@automattic/calypso-build > jest-enzyme > jest-environment-enzyme@7.1.2" has unmet peer dependency "jest@>=22.0.0".
warning "@automattic/calypso-build > jest-enzyme > jest-environment-enzyme@7.1.2" has unmet peer dependency "react@^0.13.0 || ^0.14.0 || ^15.0.0 || >=16.x".
warning "@automattic/calypso-build > enzyme-adapter-react-16 > enzyme-adapter-utils > airbnb-prop-types@2.16.0" has unmet peer dependency "react@^0.14 || ^15.0.0 || ^16.0.0-alpha".
[4/4] Building fresh packages...
success Saved lockfile.
Done in 28.54s.
yarn run v1.22.10
$ yarn run clean
$ rm -rf dist/
$ calypso-build lazy-images='./src/js/lazy-images.js' && cp ./node_modules/intersection-observer/intersection-observer.js ./dist/
Hash: 10714c4b948fdfa3cec9
Version: webpack 4.41.6
Time: 882ms
Built at: 11/30/2020 4:41:38 PM
         Asset      Size       Chunks             Chunk Names
lazy-images.js  5.39 KiB  lazy-images  [emitted]  lazy-images
Entrypoint lazy-images = lazy-images.js
[./src/js/lazy-images.js] 1.48 KiB {lazy-images} [built] [failed] [1 error]

ERROR in ./src/js/lazy-images.js
Module build failed (from ./node_modules/thread-loader/dist/cjs.js):
Thread Loader (Worker 0)
[BABEL] /jetpack/__lazy-images__clone__/src/js/lazy-images.js: Could not find plugin "proposal-class-properties". Ensure there is an entry in ./available-plugins.js for it. (While processing: "/jetpack/__lazy-images__clone__/node_modules/@babel/preset-env/lib/index.js")
    at PoolWorker.fromErrorObj (/jetpack/__lazy-images__clone__/node_modules/thread-loader/dist/WorkerPool.js:262:12)
    at /jetpack/__lazy-images__clone__/node_modules/thread-loader/dist/WorkerPool.js:204:29
    at mapSeries (/jetpack/__lazy-images__clone__/node_modules/neo-async/async.js:3625:14)
    at getPlugin (/jetpack/__lazy-images__clone__/node_modules/@babel/preset-env/lib/index.js:67:11)
    at Array.from.map.pluginName (/jetpack/__lazy-images__clone__/node_modules/@babel/preset-env/lib/index.js:258:62)
    at Array.map (<anonymous>)
    at _default (/jetpack/__lazy-images__clone__/node_modules/@babel/preset-env/lib/index.js:258:43)
    at /jetpack/__lazy-images__clone__/node_modules/@babel/helper-plugin-utils/lib/index.js:19:12
    at /jetpack/__lazy-images__clone__/node_modules/@babel/core/lib/config/full.js:199:14
    at Generator.next (<anonymous>)
    at Function.<anonymous> (/jetpack/__lazy-images__clone__/node_modules/@babel/core/lib/gensync-utils/async.js:26:3)
    at Generator.next (<anonymous>)
    at step (/jetpack/__lazy-images__clone__/node_modules/gensync/index.js:261:32)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
</details>

Is there a dependency missing or something?

**Comment by ockham** [comment]
> Ok, I think I see the path forward here.
> 
> A commit in the monorepo runs the workflow defined in .github/workflows/update-package.yml to push the commits to the mirror repo. The main work there is done by .github/actions/update-package/update-package.sh. And Github Actions even includes yarn already, so we can very easily run a build step in the middle of that. I've just pushed [1d7ffab](https://github.com/Automattic/jetpack/commit/1d7ffab6de4ca70e77f9099496455eabe71efb45) which makes that happen; details are open to discussion of course.

Ah, that sounds good! Nice spot :+1: 

> @ockham There is one problem, though: When I try `yarn build` in the package, it doesn't work.
> 
> I'm doing this in a docker environment, set up like this: [...]
> The build fails with the following output:
> [...]
> Is there a dependency missing or something?

Looks like the same error that Jeremy ran into: https://github.com/Automattic/jetpack/pull/17571#pullrequestreview-526908676

It's probably sort of a dependency issue, but I think the real problem is that we're `cd`ing into a different directory, and attempting a 'partial' installation of node modules there. In Calypso, we're using `lerna` to manage these things. I thought I'd be able to avoid introducing it here, but maybe it's the simplest way forward. I'll try something...

**Comment by ockham** [comment]
Hmm, turns out that this is actually a Babel bug: https://github.com/babel/babel/issues/12019#issuecomment-683178125:

> This issue has been fixed in [7127330](https://github.com/babel/babel/commit/7127330f02b423d9cb0555b3d2a24dbccbc149b5), `preset-env` 7.8.7.
> 
> Affected `@babel/preset-env` versions: 7.8.0 - 7.8.6.
> 
> That said, please update `@babel/preset-env` to 7.8.7 or better to update all `@babel/*` to latest versions.

**Comment by ockham** [comment]
> Hmm, turns out that this is actually a Babel bug: https://github.com/babel/babel/issues/12019#issuecomment-683178125: [../]

https://github.com/Automattic/jetpack/pull/16845 might solve that issue, but last I checked it was blocked by a cascade of other issues :grimacing:

**Comment by ockham** [comment]
> > Hmm, turns out that this is actually a Babel bug: [babel/babel#12019 (comment)](https://github.com/babel/babel/issues/12019#issuecomment-683178125): [../]
> 
> #16845 might solve that issue, but last I checked it was blocked by a cascade of other issues

To clarify: I think we need to get the `@wordpress/` monorepo upgrade (https://github.com/Automattic/jetpack/pull/16763) merged first, which should unblock the `calypso-build` upgrade (https://github.com/Automattic/jetpack/pull/16845).

**Comment by anomiex** [comment]
We've done a bunch of work since the last time someone poked at this, and in particular we redid the build system in a way that already took care of removing the minified files from the monorepo. But we can still do the part about using calypso-build, and pulling in the polyfill from node_modules instead of maintaining a copy here.

**Comment by jeherve** [comment]
The Calypso Build update has now landed, in #19374.

**Comment by anomiex** [comment]
I think the monorepo reorg made that not block this anyway.

**Review by jeherve — DISMISSED** [review]
This tests well for me. I wonder if it may be worth marking this more than a patch change since we're moving built files around. But absolutely not a blocker, up to you!

**Comment by anomiex** [comment]
The question there is whether the paths or URLs of those JS files would be considered part of the "API" of the package, or just an internal implementation detail. Let's err on the safe side.

**Review by jeherve — APPROVED** [review]

**Comment by anomiex** [comment]
r227091-wpcom

**Pull request merged**
